### PR TITLE
fix for missing columns and added entity types

### PR DESCRIPTION
**Pull request by ashwin-patil**
Fixes #

## Proposed Changes

  - Queries were failing when data source was not onboarded so added fix via columnifexists as suggested by Shain
  - added entity types IP, account and timestamp where applicable.
  -

### CODE REVIEW

File: Detections/AzureDiagnostics/KeyVaultSensitiveOperations.yaml [hunk 1]
```diff
@@ -24,7 +24,14 @@ query: |
...
+  | extend ResultType = columnifexists("ResultType", "None"), identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g = columnifexists("identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g", "None")
+  | extend id_s = columnifexists("id_s", "None"), CallerIPAddress = columnifexists("CallerIPAddress", "None"), clientInfo_s = columnifexists("clientInfo_s", "None")
+  | where ResultType != "None" and isnotempty(ResultType)
+  | where identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g != "None" and isnotempty(identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g)
+  | where id_s != "None" and isnotempty(id_s)
+  | where CallerIPAddress != "None" and isnotempty(CallerIPAddress)
+  | where clientInfo_s != "None" and isnotempty(clientInfo_s)
   | where ResourceType == "VAULTS" and ResultType == "Success" 
   | where OperationName in (SensitiveOperationList)  
   | project TimeGenerated, ResourceType, ResultType, Resource, OperationName, id_s, CallerIPAddress, identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g, requestUri_s, clientInfo_s
```
Comment (shainw) [inline Detections/AzureDiagnostics/KeyVaultSensitiveOperations.yaml]: requestUri_s is not a property that is defined.

File: Detections/AzureDiagnostics/KeyvaultMassSecretRetrieval.yaml [hunk 2]
```diff
@@ -25,15 +25,25 @@ query: |
...
+  | extend ResultType = columnifexists("ResultType", "None"), identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g = columnifexists("identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g", "None")
+  | extend id_s = columnifexists("id_s", "None"), CallerIPAddress = columnifexists("CallerIPAddress", "None"), clientInfo_s = columnifexists("clientInfo_s", "None")
+  | where ResultType != "None" and isnotempty(ResultType)
+  | where identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g != "None" and isnotempty(identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g)
+  | where id_s != "None" and isnotempty(id_s)
+  | where CallerIPAddress != "None" and isnotempty(CallerIPAddress)
+  | where clientInfo_s != "None" and isnotempty(clientInfo_s)
   | where OperationName in (OperationList)   
   ) on identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g 
   | project TimeGenerated, ResourceType, ResultType, Resource, OperationName, id_s, CallerIPAddress, identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g, requestUri_s, clientInfo_s
```
Comment (shainw) [inline Detections/AzureDiagnostics/KeyvaultMassSecretRetrieval.yaml]: requestUri_s is not a property that is defined.

File: Detections/AzureDiagnostics/TimeSeriesKeyvaultAccessAnomaly.yaml [hunk 3]
```diff
@@ -41,9 +44,16 @@ query: |
...
+  | extend ResultType = columnifexists("ResultType", "None"), identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g = columnifexists("identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g", "None")
+  | extend id_s = columnifexists("id_s", "None"), CallerIPAddress = columnifexists("CallerIPAddress", "None"), clientInfo_s = columnifexists("clientInfo_s", "None")
+  | where ResultType != "None" and isnotempty(ResultType)
+  | where identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g != "None" and isnotempty(identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g)
+  | where id_s != "None" and isnotempty(id_s)
+  | where CallerIPAddress != "None" and isnotempty(CallerIPAddress)
+  | where clientInfo_s != "None" and isnotempty(clientInfo_s)
   | where ResourceType == "VAULTS" and ResultType == "Success"
   | where OperationName in (OperationList)  
   | summarize PerOperationCount=count() by bin(TimeGenerated,1h), Resource, OperationName, id_s, CallerIPAddress, identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g, requestUri_s, clientInfo_s
```
Comment (shainw) [inline Detections/AzureDiagnostics/TimeSeriesKeyvaultAccessAnomaly.yaml]: requestUri_s is not a property that is defined.

File: Detections/AzureDiagnostics/KeyvaultMassSecretRetrieval.yaml (hunk 2 above)
Comment (ashwin-patil) [reply Detections/AzureDiagnostics/KeyvaultMassSecretRetrieval.yaml]: added fix.

File: Detections/AzureDiagnostics/KeyVaultSensitiveOperations.yaml (hunk 1 above)
Comment (ashwin-patil) [reply Detections/AzureDiagnostics/KeyVaultSensitiveOperations.yaml]: added fix.

File: Detections/AzureDiagnostics/TimeSeriesKeyvaultAccessAnomaly.yaml (hunk 3 above)
Comment (ashwin-patil) [reply Detections/AzureDiagnostics/TimeSeriesKeyvaultAccessAnomaly.yaml]: added fix.

**Pull request merged**